app/services/proforma_service/convert_exercise_to_task.rb

```diff
@@ -109,10 +139,20 @@ def exercise_files
     end
 
     def task_files_meta_data
-      exercise_files.to_h do |file|
-        # added CO- to id, otherwise the key would have CodeOcean as a prefix after export and import (cause unknown)
-        ["CO-#{file.id}", {role: file.role}]
+      # TODO: refactor?
+      task_files_hash = {
+        '@@order' => exercise_files.map {|file| "CodeOcean:CO-#{file.id}" },
+      }
+      exercise_files.each do |file|
+        task_files_hash["CodeOcean:CO-#{file.id}"] = {
+          '@@order' => ['CodeOcean:role'],
+          'CodeOcean:role' => {
+            '@@order' => ['$1'],
+            '$1' => file.role,
+          },
+        }
       end
+      task_files_hash
```

Comment:
The only refactoring I can think of is the following:

```suggestion
      task_files_hash = {
        '@@order' => [],
      }

      exercise_files.each do |file|
        task_files_hash['@@order'] << "CodeOcean:CO-#{file.id}"
        task_files_hash["CodeOcean:CO-#{file.id}"] = {
          '@@order' => ['CodeOcean:role'],
          'CodeOcean:role' => {
            '@@order' => ['$1'],
            '$1' => file.role,
          },
        }
      end

      task_files_hash
```

Most notably, I am not looping twice over the `exercise_files`, and rather add the order element when required. However, here, I am adding elements to the `task_files_hash['@@order']` within the loop, therefore requiring some expansion, potentially. 

Generally speaking, I would consider the actual performance difference, if any, to be negligible. Therefore, I would decide with readability in mind. In this sense, I slightly prefer my solution, but don't have hard feelings.